Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -144,10 +144,57 @@ static void for_each_item(VM& vm, GlobalObject& global_object, const String& nam
     }
 }
 
+// 10.4.2.3 ArraySpeciesCreate ( originalArray, length ), https://tc39.es/ecma262/#sec-arrayspeciescreate
+static Object* array_species_create(GlobalObject& global_object, Value original_array, size_t length)
+{
+    VERIFY(original_array.is_object());
```

Comment:
Can we make the parameter an `Object&` instead?

Reply:
:+1: i keep forgetting to change these runtime asserts to these compile-time restrictions, theyre so much nicer :)